packages/libs/src/sdk/services/DiscoveryNodeSelector/DiscoveryNodeSelector.ts

```diff
@@ -259,10 +264,152 @@ export class DiscoveryNodeSelector implements DiscoveryNodeSelectorService {
     return await this.select(null)
   }
 
+  public async getUniquelyOwnedEndpoints(
```

Comment:
Compare this to the current `select()` method - they're very similar... Opted to be WET for clarity but can iterate

Reply:
How hard would it be to write tests against this? The logic below seems pretty gnarly.

Reply:
Good call, will add unit tests